modules/base/rendering/renderableprism.cpp

```diff
@@ -0,0 +1,354 @@
+ *                                                                                       *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED,   *
+ * INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A         *
+ * PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT    *
+ * HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF  *
+ * CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE  *
+ * OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.                                         *
+ ****************************************************************************************/
+
+#define _USE_MATH_DEFINES
+
+#include <modules/base/rendering/renderableprism.h>
+
+#include <openspace/documentation/documentation.h>
+#include <openspace/documentation/verifier.h>
+#include <openspace/engine/globals.h>
+#include <openspace/rendering/renderengine.h>
+#include <openspace/util/updatestructures.h>
+#include <ghoul/filesystem/filesystem.h>
+#include <ghoul/logging/logmanager.h>
+#include <ghoul/opengl/openglstatecache.h>
+#include <ghoul/opengl/programobject.h>
+#include <math.h>
+#include <optional>
+
+namespace {
+    constexpr const char _loggerCat[] = "RenderablePrism";
+
+    constexpr const std::array<const char*, 2> UniformNames = {
+        "modelViewProjectionTransform", "vs_color"
+    };
+
+    constexpr openspace::properties::Property::PropertyInfo SegmentsInfo = {
+        "Segments",
+        "Segments",
+        "The number of segments the shape of the prism should have."
+    };
+
+    constexpr openspace::properties::Property::PropertyInfo LinesInfo = {
+        "NumLines",
+        "Number of Lines",
+        "The number of lines connecting the two shapes of the prism."
+    };
+
+    static const openspace::properties::Property::PropertyInfo RadiusInfo = {
+        "Radius",
+        "Radius",
+        "The radius of the prism's shape in meters."
+    };
+
+    constexpr openspace::properties::Property::PropertyInfo LineWidthInfo = {
+        "LineWidth",
+        "Line Width",
+        "This value specifies the line width."
+    };
+
+    constexpr openspace::properties::Property::PropertyInfo LineColorInfo = {
+        "Color",
+        "Color",
+        "This value determines the RGB color for the line."
+    };
+
+    constexpr openspace::properties::Property::PropertyInfo LengthInfo = {
+        "Length",
+        "Length",
+        "The length of the prism in meters."
+    };
+
+    // Generate vertices around the unit circle on the XY-plane
+    void getUnitCircleVertices(std::vector<float>& vertices, int sectorCount) {
+        float sectorStep = 2 * M_PI / sectorCount;
+        float sectorAngle;  // in radians
+
+        for (int i = 0; i < sectorCount; ++i) {
+            sectorAngle = i * sectorStep;
```

Comment:
You can define the variable inside the loop. `float sectorAngle = i * sectorStep;` 
Predefining variables before a look doesn't really do anything anymore optimization-wise as the optimizer will >99% do that for you and it leads to code that is easier to read